src/group_theory/coset.lean

```diff
@@ -170,6 +209,24 @@ instance left_rel_decidable [group α] (s : subgroup α) [d : decidable_pred (λ
   If `s` is a normal subgroup, `quotient s` is a group -/
 def quotient [group α] (s : subgroup α) : Type* := quotient (left_rel s)
 
+/-- The equivalence relation corresponding to the partition of a group by right cosets of a
+subgroup.-/
+@[to_additive "The equivalence relation corresponding to the partition of a group by right cosets of
+a subgroup."]
+def right_rel [group α] (s : subgroup α) : setoid α :=
+⟨λ x y, y * x⁻¹ ∈ s,
+  assume x, by simp [s.one_mem],
+  assume x y hxy,
+  have (y * x⁻¹)⁻¹ ∈ s, from s.inv_mem hxy,
+  by simpa using this,
+  assume x y z hxy hyz,
+  have (z * y⁻¹) * (y * x⁻¹) ∈ s, from s.mul_mem hyz hxy,
+  by simpa [mul_assoc] using this⟩
```

Comment:
I guess the formatting is copied from the `left` version. Can you make these changes there too?
```suggestion
    have (y * x⁻¹)⁻¹ ∈ s, from s.inv_mem hxy,
    by simpa using this,
  assume x y z hxy hyz,
    have (z * y⁻¹) * (y * x⁻¹) ∈ s, from s.mul_mem hyz hxy,
    by simpa [mul_assoc] using this⟩
```